src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -104,7 +104,9 @@ private Assembly TryResolveAssembly(AssemblyLoadContext context, AssemblyName as
                 {
                     foreach (var extension in _extensions)
                     {
-                        var candidatePath = Path.Combine(dependencyPath, $"{assemblyName.Name}.{extension}");
+                        var candidatePath = Path.Combine(dependencyPath,
+                            assemblyName.CultureName ?? string.Empty,
```

Comment:
If we got to this point then none of the assemblies we've already loaded match the one that is being requested. If one of them did then we wouldn't have been called at all. Also, we don't want to load the same assembly twice. So we skip the already loaded one and go looking for another.

I can't recall the details about when this would happen, but there was a reason for it.